17.1 Dungeon - Odin Past.

taig40 opened the pull request:
<!-- NOTE: Anything within these brackets will be hidden on the preview of the Pull Request. -->

* **Addressed Issue(s)**:
Partially #6469 

<!--
Please specify the rAthena [GitHub issue(s)](https://help.github.com/articles/autolinked-references-and-urls/#issues-and-pull-requests) this pull request amends.
If no issue exists yet, please [create one](https://github.com/rathena/rathena/issues/new) first and then link your pull request to the amendment!
-->

* **Server Mode**: 
Renewal

<!-- Which mode does this pull request apply to: Pre-Renewal, Renewal, or Both? -->

* **Description of Pull Request**:
Add warps for Odin Past, at Odin Temple F3, add merchants for enchanting, add mobs, mob skills, spawns and items relative to that update.

<!-- Describe how this pull request will resolve the issue(s) listed above. -->


Todos:
- [x] Check official enchantment NPC script
- [x] Monsters (custom)
- [x] Monster skills (custom)
- [ ] https://github.com/rathena/rathena/pull/6753

taig40 committed:
Commit

taig40 committed:
Update scripts_athena.conf

Add my script to scripts athena.

taig40 committed:
Update odin_enchanter.txt

Add the header

JohnnyPlayy commented:
https://youtu.be/UMr51B9DRO0
!

mundussan commented:
> https://youtu.be/UMr51B9DRO0
> !

I'll check this video when I get home.
Thank you!

Atemo commented on npc/re/mobs/dungeons/odin.txt:
```diff
@@ -54,3 +54,18 @@ odin_tem03,0,0	monster	Valkyrie	1765,2,5400000,1800000
+odin_past,0,0,0,0	boss_monster	Valkyrie Reginleif	20610,1,28800000,600000,1
+odin_past,0,0,0,0	boss_monster	Valkyrie Ingrid	20611,1,28800000,600000,1
+odin_past,0,0	monster	Blue Plant	1079,18,1800000,900000
+odin_past,0,0	monster	Shining Plant	1083,18,1800000,900000
```
The spawn number is different from https://www.divine-pride.net/database/map/odin_past
Do you mind to share your source?

mundussan replied:
Hmmm, well, I didn't realized which spawn was wrong. I just copied the spawn from Odin Temp 03. I'll fix it according to DP

Balferian replied:
Not sure where is correct info, but korean navi luas contain next data:
There are few differences in amount of mobs.
(respawn time is dummy)
> odin_past,0,0	monster	SHINING_PLANT	1083,2,5000,0
odin_past,0,0	monster	BLUE_PLANT	1079,5,5000,0
odin_past,0,0	monster	ANGELGOLT	20604,40,5000,0
odin_past,0,0	monster	ANGELGOLT2	20605,60,5000,0
odin_past,0,0	monster	PLASMA_SPT	20609,5,5000,0
odin_past,0,0	monster	PLASMA_ARCH	20608,5,5000,0
odin_past,0,0	monster	HOLY_FRUS	20606,45,5000,0
odin_past,0,0	monster	HOLY_SKOGUL	20607,45,5000,0
odin_past,0,0	boss_monster	REGINLEIF	20610,1,10800000,600000,0
odin_past,0,0	boss_monster	INGRID	20611,1,10800000,600000,0

mundussan replied:
Seems to be correct. I'll fix it when I get home.

Atemo commented on npc/re/merchants/odin_enchanter.txt:
```diff
@@ -0,0 +1,367 @@
+//= 1.0 First release. [mundussan]
+//============================================================ 
+
+hu_in01,97,323,4	script	Susan Grey#enchanter	893,{
```
Are your npcs from replay?

mundussan replied:
If with replay you mean taken from someone else, no. I used some parts from actual rAthena scripts and the rest I made by myself.

mundussan replied:
Oh wait, if you mean the conversation of the npc, then yes, its from replay.
I can't play/understand kRO, so I took the dialogue from NovaRO.

mundussan replied:
> .

I see. I don't have a replay recording from KRO. As I said, I used NovaRO dialogue to make my npc, and then I did my logic into it.

Shaktohh replied:
You can use Google Lens to translate from your camera's phone directly :)

JohnnyPlayy commented:
The last time I entered odin past the message that appeared when you had no level was this.

`message strcharinfo(0), strcharinfo(0) + " 엘레나 : 뭐지? 이 빛은...";`

mundussan commented:
> 엘레나 : 뭐지? 이 빛은...";

Can you record a replay if you have access to kRO, please?
Or do you know who is Elena?

taig40 committed:
Update odin.txt

Correct the mob spawn according to Divine Pride.

JohnnyPlayy commented:
I had access on sakray I even had a replay! however he was corrupted...

mundussan commented:
> 

Ohhhhhhhhhh :(
That's sad.

JohnnyPlayy commented:
https://www.pastiebin.com/61e4382a1b986

This information I took from sakray as soon as it came out! But something must have changed. And that script is incomplete. Because when you first click on the portal, a circle of NPCs appears around you!

mundussan commented:
Oh, good to know.
I'll update mine accordingly.
Thanks, Johnny.
I wish I could play on kRO's sakray too :'(

JohnnyPlayy commented:
All accounts on sakray have been reset and now only developers have free access

mundussan commented:
Also, @JohnnyPlayy if you got any info about the enchantment npc and wish to share, it would be amazing.

JohnnyPlayy commented:
![image](https://user-images.githubusercontent.com/17951544/149667515-8188299f-c46b-49ab-8010-e18bc0527da5.png)
The only problem is that I'm out of level and out of items

JohnnyPlayy commented:
https://www.pastiebin.com/61e446b9b26eb

Only the part of the enchantment function was missing!

JohnnyPlayy commented:
Delete your old script and use mine as a base! And leave it in Korean!

mundussan commented:
Sure.
Thanks for the info!

taig40 committed:
Some cleanup

Applied JohnnyPlayy's suggestions and some cleanup at code.

taig40 committed:
Merge branch 'master' into hotfix/issue6469

taig40 committed:
Fix Appveyor error

taig40 committed:
Merge branch 'hotfix/issue6469' of https://github.com/mundussan/rathena into hotfix/issue6469

taig40 committed:
Small typo

taig40 committed:
Merge branch 'master' into hotfix/issue6469

ryomahostingph commented:
can't merge in https://github.com/rathena/rathena/commit/08192a35bc856b38e94aa0d8b0c87574008cd97e
up

taig40 committed:
Merge branch 'master' into hotfix/issue6469

taig40 committed:
Fixed small typo

taig40 committed:
Another small cleanup

Everade commented:
Are all monsters, monster skills as well as item stats/settings custom or are they verified?

mundussan commented:
> Are most monsters, monster skills as well as item stats/settings custom or are they verified? Just asking because some data looks off, and divine lacks some info. In case they're custom, please add a todo notice.

Most of them are custom, because I didn't found any official source of them.
Specially monster ATK/MATK and skills rates.

Everade reviewed (commented):
General review.

thywdasot commented:
thanks.

thywdasot commented:
#define MAX_SKILL 1454 Gets conflict with athena web-service, i suggest keep on 1454, but idk how impact that.

thywdasot commented on src/common/mmo.hpp:
```diff
@@ -83,7 +83,7 @@ typedef uint32 t_itemid;
 #define MAX_BANK_ZENY SINT32_MAX ///Max zeny in Bank
 #define MAX_FAME 1000000000 ///Max fame points
 #define MAX_CART 100 ///Maximum item in cart
-#define MAX_SKILL 1454 ///Maximum skill can be hold by Player, Homunculus, & Mercenary (skill list) AND skill_db limit
```
#define MAX_SKILL 1455 Gets conflict with athena web-service of vstrumpf
i reproduce and skill hotkeys not saving.

Lemongrass3110 replied:
This is not related to this PR and I think you did something wrong.

Everade commented:
@mundussan Please note the a review isn't anything personal but here to improve and/or fix the content for everyone. It isn't always easy to verify kRO data, that's why todo notices are helpful to fix things up later down the road. If you feel that i've overdone it, adding the todo message to the pull description works too if we can fix them before merging to master. The requested changes are very basic and shouldn't take much of your time.

mundussan commented:
> @mundussan Please note the a review isn't anything personal but here to improve and/or fix the content for everyone. It isn't always easy to verify kRO data, that's why todo notices are helpful to fix things up later down the road. If you feel that i've overdone it, adding the todo message to the pull description works too if we can fix them before merging to master. The requested changes are very basic and shouldn't take much of your time.

I closed it because I don't have time now to fix it (the mob data, and the scripts).
Nothing personal, and I know what is the purpose of your reviews.
Well, you can let it open tho, but I'll only fix it when I could

Commit:
Fixes from general review
Script cleanup/minor improvements
Corrected indentation issues
Corrected some item and monster data
Moved warp script to Renewal

Commit:
Merge remote-tracking branch 'remotes/rathena/master' into hotfix/issue6469

# Resolved Conflicts:
#	db/re/mob_skill_db.txt

Commit:
Merge remote-tracking branch 'remotes/rathena/master' into hotfix/issue6469

# Resolved Conflicts:
#	db/re/item_group_db.yml
#	db/re/mob_skill_db.txt
#	db/re/skill_db.yml
#	src/map/itemdb.hpp
#	src/map/script_constants.hpp

Commit:
Minor improvements on enchanter
Some translation improvements
Added randomoption check
Reduced Silvervice Fruit cost for reset
Corrected NPC location

Commit:
Corrected random option check

Lemongrass3110 committed:
Update odin_enchanter.txt

Lemongrass3110 committed:
Merge branch 'master' of https://github.com/rathena/rathena into hotfix/issue6469

# Conflicts:
#	db/re/item_group_db.yml
#	src/map/itemdb.hpp
#	src/map/script_constants.hpp

Atemo committed:
Small clean-up

Atemo committed:
Changed item rate to more realistic values (rates still unknown)

Pokye commented on npc/re/mobs/dungeons/odin.txt:
```diff
@@ -54,3 +51,17 @@ odin_tem03,0,0	monster	Valkyrie	1765,2,5400000,1800000
+odin_past,0,0	monster	Holy Skogul	20607, 45,5000
+odin_past,0,0	monster	Arch Plasma	20608,5,5000
+odin_past,0,0	monster	Spectral Plasma	20609,5,5000
+odin_past,0,0,0,0	boss_monster	Valkyrie Reginleif	20610,1,28800000,600000,1
```
The mpvs Reginleif and Ingrid aren't they randomly summoned after killing a certain amount of Holy Frus and Holy Skogul?

Pokye replied:
**Valkyrie Ingrid** appears after the entire server has eliminated 300 Holy Skogul. The recount is done every 6 hours.
**Valkyrie Reginleif** appears after the entire server has eliminated 300 Holy Frus. The recount is done every 6 hours.

official information from bRO. - https://browiki.org/wiki/MVP#cite_ref-6

IAmJennyCookz commented:
Any update?

AsurielRO commented:
I believe [New mobs skills](https://github.com/rathena/rathena/pull/6753) Have been implemented.

Is there anything left on this before the merge?

eppc0330 commented:
@AsurielRO Field boss(Reginleif and Ingrid) trigger is incomplete yet.
About #6753, yes skills are all implemented already.
